### ISSUE TITLE
iOS - Code 0 (read and subscribe error)

### ISSUE THREAD

**Issue by tfmontague** [randdusing/cordova-plugin-bluetoothle#278]

**Comment by tfmontague** [comment]
This error also looks like it's related to notifications and `setNotify`.

**Error Domain=CBErrorDomain Code=0**
http://stackoverflow.com/questions/9545500/notifications-on-corebluetooth-returning-a-cberrordomain-code-0

> I think some of these errors occur when you try to read a value that is marked to only be available through notifications rather than read directly.

**Comment by tfmontague** [comment]
Setting `notify: true` in properties seems to fix this error.

```
var params = {
  service: "1234",
  characteristics: [{
    uuid: "ABCD",
    permissions: {
      read: true
    },
    properties: {
      read: true,
      notify: true
    },
    value: bluetoothle.bytesToEncodedString(
      bluetoothle.stringToBytes("characterValue"))
  }]
};

bluetoothle.addService(..., ..., params);
```

**Comment by randdusing** [comment]
How often are you changing the characteristics on the server side? I wonder if the client is caching the characteristics' configurations and causing issues.

**Comment by tfmontague** [comment]
If you mean "changing" as editing the code, the answer is frequently. But adding `notify: true` fixed the issue right away. Didn't seem like it was cached.

**Comment by randdusing** [comment]
Were you trying to read or subscribe from the client side?

**Comment by tfmontague** [comment]
I have a client app with the code needed to read or subscribe. If I want to read I comment out the subscribe call. If I want to subscribe I comment out the read call.

I got the Code 0 error for both `read` and `subscribe` on different builds of the client app. It went away for `subscribe` after adding `notify: true`; for `read` it never went away (maybe due to the hard-coded requestId).

All three of these read errors ( #278, #241, #275 ) may have been caused by hard-coding the `requestId` for `respond`. I've fixed the RequestId so it's no longer hard-coded, but I can't verify if that fixed the read errors. Haven't spent enough time with the app after making this code change to conclude. But I will try to give an update in a day or two.
